sklearn/linear_model/_linear_loss.py

```diff
@@ -0,0 +1,368 @@
+"""
+Loss functions for linear models with raw_prediction = X @ coef
+"""
+import numpy as np
+from scipy import sparse
+from ..utils.extmath import squared_norm
+
+
+class LinearLoss:
+    """General class for loss functions with raw_prediction = X @ coef.
+
+    The loss is the sum of per sample losses and includes an L2 term::
+
+        loss = sum_i s_i loss(y_i, X_i @ coef + intercept)
+               + 1/2 * l2_reg_strength * ||coef||_2^2
+
+    with sample weights s_i=1 if sample_weight=None.
+
+    Gradient and hessian, for simplicity without intercept, are::
+
+        gradient = X.T @ loss.gradient + l2_reg_strength * coef
+        hessian = X.T @ diag(loss.hessian) @ X + l2_reg_strength * identity
```

Comment:
`alpha` is also used as the generic name for total regularization in elastic net (l2 + l1) that this then split in 2 sub-components for the two terms based on `l1_ratio`.

It's just that I find that using Greek letters in code variables was a bad idea that makes the code confusing to read, especially for newcomers.  We cannot change the scikit-learn public API (because of backward compat and the huge amound of third party doc, tutorial, books, MOOC videos that would be rendered obsolete by such a change) but for new, private code we can use more explicit variable names if we want.

Reply:
OK, thank you for this remark.